sycl/test/basic_tests/parallel_for_range.cpp

```diff
@@ -0,0 +1,93 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+int main() {
+  auto asyncHandler = [](exception_list es) {
+    for (auto& e : es) {
+      std::rethrow_exception(e);
+    }
+  };
+
+  // parallel_for, 100 global, 3 local -> fail.
+  try {
+    queue q(asyncHandler);
+    q.submit([&](handler &cgh) {
+        cgh.parallel_for<class a>(nd_range<1>(range<1>(100), range<1>(3)),
+            [=](nd_item<1> id) {});
+    });
+    q.wait_and_throw();
+    assert(false && "Should have thrown exception");
+  } catch (...) {
```

Comment:
Please add here only the exception we are expecting.
And I would like to very precise, such as `cl::sycl::runtime_error`, to be sure it is not https://en.cppreference.com/w/cpp/error/runtime_error